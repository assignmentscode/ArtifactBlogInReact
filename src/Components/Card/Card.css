body, html {
    height: 100%;
    width: 100%;
    font-weight: 10;
    margin: 0px;
  }
.card {
    /* max-width: 280px; */
    flex-basis: 300px;
    align-self: flex-start;
    text-align: center;
    font-family: arial;
    border : 0.2px solid gray;
    margin-top: 2%;
    margin-bottom: 2%;
    margin-left: auto;
    margin-right: auto;
  }
  .blog-image {
    width : 300px;
    height: 220px;
}
.date {
    margin: 0;
    float: left;
    font-size: 12px;
    color: gray;
  }
  .read-time {
    margin :0;
    float:right;
    font-size: 12px;
    color: gray;
  }
  .post-heading {
    font-size: 20px;
    font-weight: 500;
    color: black;
    padding-left: 8px;
    padding-right: 8px;
    margin-bottom: 8px;
    text-align: left;
}
.blog-content {
    font-size: 14px;
    color: gray;
    text-align: left;
    margin: 2px 8px;
}
figcaption {
    margin: 5px;
    padding-bottom: 8px;
}
.clap-logo {
    float: left;
    padding: 1px 5px 5px 5px;
    width: 8%;
}
.heart-logo {
    float: right;
    padding: 3px 10px 10px 5px;
    width: 8%;
}
.clap-count {
    float:left;
    padding: 3px;
    color: gray;
}
hr {
    margin: 8px 10px;
}