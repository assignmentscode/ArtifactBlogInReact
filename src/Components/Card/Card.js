import React, { Component } from 'react';
import './Card.css';

class Card extends Component {
    render(){
        const clapIcon = require('./../../Icons/clapping.svg');
        const blackHeart = require('./../../Icons/heart-black.svg');
        const redHeart = require('./../../Icons/heart-red.svg');
        const imagePath = require('./../../Images/'.concat(this.props.data.image));
        const dateData = this.props.data.date;
        const readingTimeData = this.props.data.readingTime;
        const titleData = this.props.data.title;
        const descriptionData = this.props.data.description;
        const clapsData = this.props.data.claps;
        const likedData = this.props.data.liked;

        return(
            <div className="card">
                <img src={imagePath} alt="John" className="blog-image"/>
                <figcaption>
                    <span className="date">{dateData}</span>
                    <span className="read-time">{readingTimeData}</span>
                </figcaption>
                <p className="post-heading">{titleData}</p>
                <p className="blog-content">{descriptionData}</p>
                <hr/>
                <img src={clapIcon} alt="Clap" className="clap-logo"/>
                <span className="clap-count">{clapsData}</span>
                <img src={(likedData===true)?redHeart:blackHeart} alt="Heart" className="heart-logo"/>
            </div>
        );
  }
}

export default Card;