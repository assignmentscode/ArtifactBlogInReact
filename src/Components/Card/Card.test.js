import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Card from './Card';
const BlogData = require('./../../mockData/index.json');

describe('Card', ()=> {
    it('should render one exactly card', ()=> {
        const tree = renderer.create(<Card data={BlogData[0]}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    const wrapper = shallow(<Card data={BlogData[0]}/>);
    it('should not have any props to be empty', ()=> {
        // console.log(wrapper.instance().props.data);
        const propData = wrapper.instance().props.data;
        expect(propData).toBeDefined();
    });
    it('should have 8 child of div element', () => {
        // console.log(wrapper.find('div').getElement().props.children.length);
        expect(wrapper.find('div').getElement().props.children.length).toEqual(8);
    });
});