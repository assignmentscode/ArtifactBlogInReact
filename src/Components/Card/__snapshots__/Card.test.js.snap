// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render one exactly card 1`] = `
<div
  className="card"
>
  <img
    alt="John"
    className="blog-image"
    src="abstract.png"
  />
  <figcaption>
    <span
      className="date"
    >
      2nd Januray, 2018
    </span>
    <span
      className="read-time"
    >
      2 mins
    </span>
  </figcaption>
  <p
    className="post-heading"
  >
    The future of abstract art and the culture ...
  </p>
  <p
    className="blog-content"
  >
    Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
  </p>
  <hr />
  <img
    alt="Clap"
    className="clap-logo"
    src={
      Object {
        "ReactComponent": [Function],
        "__esModule": true,
        "default": "clapping.svg",
      }
    }
  />
  <span
    className="clap-count"
  >
    10
  </span>
  <img
    alt="Heart"
    className="heart-logo"
    src={
      Object {
        "ReactComponent": [Function],
        "__esModule": true,
        "default": "heart-black.svg",
      }
    }
  />
</div>
`;
